chains/celo/listener.go

```diff
@@ -33,3 +44,28 @@ func (l *listener) start() error {
 func (l *listener) close() {
 	l.conn.Close()
 }
+
+func (l *listener) getTransactionBlockHash(hash common.Hash) (blockHash common.Hash, err error) {
+
+	receipt, err := l.conn.Client().TransactionReceipt(context.Background(), hash)
+	if err != nil {
+		return hash, fmt.Errorf("unable to get BlockHash: %s", err)
+	}
+	return receipt.BlockHash, nil
+}
+
+func (l *listener) getBlockTransactionsByHash(hash common.Hash) (txHashes []common.Hash, txRoot common.Hash, err error) {
+	block, err := l.conn.Client().BlockByHash(context.Background(), hash)
```

Comment:
IIRC there's an option on this RPC call that allows you to return just an array of hashes instead of full transaction objects so you won't have to iterate over all of the objects to pull the hashes later in your code!

Reply:
I saw that parameter too in the docs: https://eth.wiki/json-rpc/API#eth_getblockbyhash  
Geth allow us to set that parameter though 😢 
```
func (ec *Client) BlockByHash(ctx context.Context, hash common.Hash) (*types.Block, error) {
	return ec.getBlock(ctx, "eth_getBlockByHash", hash, true)
}
```